From 1e12a153d9a82a771c3bfd95c0265b810a424b3c Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Mon, 1 Jul 2013 14:41:33 +0200
Subject: [PATCH] patch: lua-5.2

Squashed commit of the following:

commit 96f65d260c6e40940f2818b721c19565c1b40607
Author: Vincent Untz <vuntz@gnome.org>
Date:   Wed Jan 11 09:52:25 2012 +0100

    Fix build with lua 5.2 by not using API deprecated in 5.1 already

    https://bugzilla.gnome.org/show_bug.cgi?id=667675
    (cherry picked from commit a14a29c39352c60f003a65b721c9af8a1d8d20df)
---
 operations/workshop/external/gluas.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/operations/workshop/external/gluas.c b/operations/workshop/external/gluas.c
index 8ba1101..63e82a2 100644
--- a/operations/workshop/external/gluas.c
+++ b/operations/workshop/external/gluas.c
@@ -97,7 +97,7 @@ static int l_progress  (lua_State * lua);
 static int l_flush     (lua_State * lua);
 static int l_print     (lua_State * lua);
 
-static const luaL_reg gluas_functions[] =
+static const luaL_Reg gluas_functions[] =
 {
     {"set_rgba",    l_set_rgba},
     {"get_rgba",    l_get_rgba},
@@ -122,7 +122,7 @@ static const luaL_reg gluas_functions[] =
 };
 static void
 register_functions (lua_State      *L,
-                    const luaL_reg *l)
+                    const luaL_Reg *l)
 {
   for (;l->name; l++)
     lua_register (L, l->name, l->func);
@@ -146,7 +146,7 @@ drawable_lua_process (GeglOperation       *op,
     lua_State *L;
     Priv p;
 
-    L = lua_open ();
+    L = luaL_newstate ();
     luaL_openlibs (L);
 
     register_functions (L, gluas_functions);
-- 
1.8.3.1

